### DISCUSSION TITLE
'Howto' save results

### DISCUSSION THREAD

**Discussion by wakawakka** [NVlabs/instant-ngp#491] [Q&A]
Hello,
i have 2 questions.

1. Are there ways to save the results? (if we don't want to use mesh export) (for eg. if I want to use created scene statically in my game).
2. Is there documentation somewhere to quickly figure it out with API python usage ngp?

**Comment by rozeappletree** [comment]
Following the thread

**Comment by Tom94** [comment]
Hi there, the answer is unfortunately no. This codebase does not allow exporting 3D assets that can be used in, for example, a game, other than the rudimentary marching cubes based meshes. A more advanced NeRF->mesh conversion would have to be programmed / engineered on top of this codebase (e.g. based on one of the latest NeRF->mesh research papers).

**Reply by rozeappletree** [reply]
![IMG_20220419_083525](https://user-images.githubusercontent.com/45640029/163983142-85266557-5b9d-479f-a386-8b43f7c7cefe.jpg)
Then what does this `--save-mesh` flag do exactly?

**Reply by Tom94** [reply]
Those are the "rudimentary marching cubes based meshes" I referred to. OP asked for something different from those.

**Reply by rozeappletree** [reply]
Oh so unfortunate. Thanks for the help!

**Comment by YuhsiHu** [comment]
hello,  could you please tell me how to save images in GUI or in command line? For example, I want to save image using the camera in base_cam.json, so I ran:

`python scripts/run.py --scene data/nerf/fox/ --mode nerf --load_snapshot data/nerf/fox/base.msgpack --screenshot_transforms data/nerf/fox/base_cam.json --screenshot_dir data/nerf/fox/screenshot`

but there is nothing in folder, why?

**Reply by aritra-pal** [reply]
I think you need to specify --screenshot_w and --screenshot_h

**Reply by YuhsiHu** [reply]
Thank you! Now it can work!
